### PR TITLE
zlint 3.4.1 (new formula)

### PR DESCRIPTION
**Pull request by bpanesar**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/zlint.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "go" => :build
+
+  def install
+    system "make", "--directory=v3", "GIT_VERSION=v#{version}", "zlint"
+    bin.install "v3/zlint"
+  end
+
+  test do
+    assert_match "ZLint version v#{version}",
+      shell_output("#{bin}/zlint -version")
```
Comment (chenrui333) [inline Formula/zlint.rb:18]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (bpanesar) [reply Formula/zlint.rb:18]: Added a test to have zlint lint a certificate.

**Comment by bpanesar** [comment]
@chenrui333, should I squash my second commit?

File: Formula/zlint.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      "BnRf8OBu55C7fsCHRayljjW0IpyZCjCjUzBRMB0GA1UdDgQWBBRIDxULqVXg4e4Z",
+      "+3QzKRG4UpfiFjAfBgNVHSMEGDAWgBRIDxULqVXg4e4Z+3QzKRG4UpfiFjAPBgNV",
+      "HRMBAf8EBTADAQH/MAoGCCqGSM49BAMDA2gAMGUCMQCCH1Sb8Hda7CttzpQjUb9c",
+      "VuG4qtyprIRvovHqrpRT0fYUJW9YmtUAHFTGijzwZSsCMAQXEXsFblfY4cwX3DYR",
+      "qg5yjd3Syn4cBMqtHKVYyi5s+6+vOONWPYBn9re7oF56jQ==",
+      "-----END CERTIFICATE-----",
+    ].join("\n") + "\n"
+    File.write("cert.pem", cert)
+
+    shell_output("#{bin}/zlint -longSummary cert.pem")
```
Comment (ZhongRuoyu) [inline Formula/zlint.rb:36]:
```suggestion
    (testpath/"cert.pem").write <<~EOS
      -----BEGIN CERTIFICATE-----
      MIIBzjCCAVSgAwIBAgIUUS5gI8Zv+vKmMsc4JnKpXoNUfXIwCgYIKoZIzj0EAwMw
      HjEcMBoGA1UEAwwTaG9tZWJyZXctemxpbnQtdGVzdDAeFw0yMjAxMDIwNDAwMjNa
      Fw0yMzAxMDIwNDAwMjNaMB4xHDAaBgNVBAMME2hvbWVicmV3LXpsaW50LXRlc3Qw
      djAQBgcqhkjOPQIBBgUrgQQAIgNiAASn3DDorzrDQiWnnXv0uS722si9zx1HK6Va
      CXQpHm/8t1SmMEYdVIU4j5UzbVKpoMIkk9twC3AiDUVZwdBNL2rqO8smZiKOh0Tz
      BnRf8OBu55C7fsCHRayljjW0IpyZCjCjUzBRMB0GA1UdDgQWBBRIDxULqVXg4e4Z
      +3QzKRG4UpfiFjAfBgNVHSMEGDAWgBRIDxULqVXg4e4Z+3QzKRG4UpfiFjAPBgNV
      HRMBAf8EBTADAQH/MAoGCCqGSM49BAMDA2gAMGUCMQCCH1Sb8Hda7CttzpQjUb9c
      VuG4qtyprIRvovHqrpRT0fYUJW9YmtUAHFTGijzwZSsCMAQXEXsFblfY4cwX3DYR
      qg5yjd3Syn4cBMqtHKVYyi5s+6+vOONWPYBn9re7oF56jQ==
      -----END CERTIFICATE-----
    EOS
    out = shell_output("#{bin}/zlint -longSummary cert.pem")
    %w[
      e_ca_organization_name_missing
      e_root_ca_key_usage_present
      e_ca_country_name_missing
      e_ca_key_usage_missing
    ].each do |err|
      assert_match err, out
    end
```

Comment (bpanesar) [reply Formula/zlint.rb]: Applied this change with a slight variation on the cert used and lint results.

File: Formula/zlint.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+class Zlint < Formula
+  desc "X.509 Certificate Linter focused on Web PKI standards and requirements"
+  homepage "https://github.com/zmap/zlint"
+  url "https://github.com/zmap/zlint.git",
+      tag:      "v3.4.1",
+      revision: "84da589869bc2d096b7ae42606bf75c602d88806"
+  license "Apache-2.0"
+
```
[the remark this answers is not in the document]
Comment (bpanesar) [reply Formula/zlint.rb:9]: Added.

**Review by chenrui333 — APPROVED** [review]

**Comment by ZhongRuoyu** [comment]
Thanks for your first contribution to Homebrew, @bpanesar!

**Pull request closed without merging**